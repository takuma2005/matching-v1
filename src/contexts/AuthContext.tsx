import React, { createContext, useContext, useState, useEffect, ReactNode } from 'react';

import { getApiClient, API_CONFIG, mockStudents } from '@/services/api/mock';
import { User, Student } from '@/services/api/types';

interface AuthContextType {
  user: User | null;
  student: Student | null;
  role: 'student' | 'tutor' | null;
  isLoading: boolean;
  needsProfileCompletion: boolean;
  signIn: (userData: MockSignInInput) => Promise<boolean>;
  signUp: (email: string, password: string, userData: Partial<User>) => Promise<boolean>;
  signOut: () => Promise<void>;
  refreshUser: () => Promise<void>;
  updateProfile: (profileData: Partial<User>) => Promise<boolean>;
  switchRole: (
    newRole: 'student' | 'tutor',
  ) => Promise<{ success: boolean; needsCompletion?: boolean }>;
  completeProfile: () => void;
}

const AuthContext = createContext<AuthContextType | undefined>(undefined);

export function useAuth() {
  const context = useContext(AuthContext);
  if (context === undefined) {
    throw new Error('useAuth must be used within an AuthProvider');
  }
  return context;
}

interface AuthProviderProps {
  children: ReactNode;
}

// モックログイン用の入力型（電話番号ベースの擬似サインイン）
type MockSignInInput = {
  id: string;
  role: 'student' | 'tutor';
  phoneNumber: string;
  name: string;
  age?: string | number;
  grade?: string;
  email?: string;
};

const DEFAULT_MOCK_PASSWORD = 'password123';

export function AuthProvider({ children }: AuthProviderProps) {
  const [user, setUser] = useState<User | null>(null);
  const [student, setStudent] = useState<Student | null>(null);
  const [role, setRole] = useState<'student' | 'tutor' | null>(null);
  const [isLoading, setIsLoading] = useState(true);
  const [needsProfileCompletion, setNeedsProfileCompletion] = useState(false);

  const apiClient = React.useMemo(() => getApiClient(), []);

  // 初期化時に現在のユーザー情報を取得
  useEffect(() => {
    const initAuth = async () => {
      try {
        const response = await apiClient.auth.getCurrentUser();
        if (response.success && response.data) {
          setUser(response.data);
          setRole(response.data.role);

          // 生徒プロフィールも取得
          const studentResponse = await apiClient.student.getProfile(response.data.id);
          if (studentResponse.success) {
            setStudent(studentResponse.data);
          }
        } else if (API_CONFIG.USE_MOCK) {
          // モック環境ではデフォルトユーザーで自動ログイン
          const def = mockStudents?.[0];
          if (def) {
            const mockUser: User = {
              id: def.id,
              email: def.email,
              name: def.name,
              role: 'student',
              created_at: def.created_at,
              updated_at: def.updated_at,
            };
            setUser(mockUser);
            setRole('student');
            const studentResponse = await apiClient.student.getProfile(mockUser.id);
            if (studentResponse.success) {
              setStudent(studentResponse.data);
            }
          }
        }
      } catch (error) {
        console.error('認証初期化エラー:', error);
      } finally {
        setIsLoading(false);
      }
    };

    initAuth();
  }, [apiClient]);

  const signIn = React.useCallback(
    async (userData: MockSignInInput): Promise<boolean> => {
      try {
        setIsLoading(true);

        const sanitizedPhone = userData.phoneNumber.replace(/[^0-9]/g, '');
        const email =
          userData.email ||
          (sanitizedPhone ? `${sanitizedPhone}@example.com` : `${userData.id}@example.com`);

        let response = await apiClient.auth.signIn(email, DEFAULT_MOCK_PASSWORD);

        if (!response.success || !response.data) {
          const signUpResponse = await apiClient.auth.signUp(email, DEFAULT_MOCK_PASSWORD, {
            name: userData.name,
            role: userData.role,
          });

          if (!signUpResponse.success || !signUpResponse.data) {
            return false;
          }

          response = signUpResponse;
        }

        const authenticatedUser = response.data;
        setUser(authenticatedUser);
        setRole(authenticatedUser.role);

        if (authenticatedUser.role === 'student') {
          const studentResponse = await apiClient.student.getProfile(authenticatedUser.id);
          if (studentResponse.success) {
            setStudent(studentResponse.data);
          }
        } else {
          setStudent(null);
        }

        return true;
      } catch (error) {
        console.error('サインインエラー:', error);
        return false;
      } finally {
        setIsLoading(false);
      }
    },
    [apiClient],
  );

  const signUp = React.useCallback(
    async (email: string, password: string, userData: Partial<User>): Promise<boolean> => {
      try {
        setIsLoading(true);
        const response = await apiClient.auth.signUp(email, password, userData);

        if (response.success && response.data) {
          setUser(response.data);
          setRole(response.data.role);

          if (response.data.role === 'student') {
            const studentResponse = await apiClient.student.getProfile(response.data.id);
            if (studentResponse.success) {
              setStudent(studentResponse.data);
            }
          }
          return true;
        }

        return false;
      } catch (error) {
        console.error('サインアップエラー:', error);
        return false;
      } finally {
        setIsLoading(false);
      }
    },
    [apiClient],
  );

  const signOut = React.useCallback(async (): Promise<void> => {
    try {
      setIsLoading(true);
      const response = await apiClient.auth.signOut();
      if (response.success) {
        setUser(null);
        setStudent(null);
        setRole(null);
        setNeedsProfileCompletion(false);
      }
    } catch (error) {
      console.error('サインアウトエラー:', error);
    } finally {
      setIsLoading(false);
    }
  }, [apiClient]);

  const refreshUser = React.useCallback(async (): Promise<void> => {
    if (!user) return;

    try {
      const response = await apiClient.auth.getCurrentUser();
      if (response.success && response.data) {
        setUser(response.data);
        setRole(response.data.role);

        const studentResponse = await apiClient.student.getProfile(response.data.id);
        if (studentResponse.success) {
          setStudent(studentResponse.data);
        }
      }
    } catch (error) {
      console.error('ユーザー情報更新エラー:', error);
    }
  }, [user, apiClient]);

  const updateProfile = React.useCallback(
    async (profileData: Partial<User>): Promise<boolean> => {
      if (!user) return false;

      try {
        setIsLoading(true);
        // Mock implementation - in real app would call API
        const updatedUser = { ...user, ...profileData, updated_at: new Date().toISOString() };
        setUser(updatedUser);
        return true;
      } catch (error) {
        console.error('プロフィール更新エラー:', error);
        return false;
      } finally {
        setIsLoading(false);
      }
    },
    [user],
  );

  const switchRole = React.useCallback(
    async (
      newRole: 'student' | 'tutor',
    ): Promise<{ success: boolean; needsCompletion?: boolean }> => {
      if (!user) return { success: false };

      try {
        setIsLoading(true);

        // ユーザーの役割を更新
        const updatedUser = { ...user, role: newRole, updated_at: new Date().toISOString() };
        setUser(updatedUser);
        setRole(newRole);

        // 新しい役割でのプロファイル情報をチェック

        if (newRole === 'student') {
          // 学生プロファイルを確認/作成
          const studentResponse = await apiClient.student.getProfile(user.id);
          if (!studentResponse.success) {
            // 学生プロファイルが存在しない場合、最小限のプロファイルを作成
            const newStudent: Student = {
              id: user.id,
              name: user.name,
              email: user.email,
              role: 'student',
              age: 18,
              grade: '',
              subjects_interested: [],
              coins: 0,
              created_at: new Date().toISOString(),
              updated_at: new Date().toISOString(),
            };
            setStudent(newStudent);
            setNeedsProfileCompletion(true);
            return { success: true, needsCompletion: true };
          } else {
            setStudent(studentResponse.data);
          }
        } else if (newRole === 'tutor') {
          // 先輩プロファイルを確認
          // TODO: Tutor API実装後に適切に処理
          setNeedsProfileCompletion(true);
          return { success: true, needsCompletion: true };
        }

        return { success: true, needsCompletion: false };
      } catch (error) {
        console.error('役割切り替えエラー:', error);
        return { success: false };
      } finally {
        setIsLoading(false);
      }
    },
    [user, apiClient.student],
  );

  const completeProfile = React.useCallback(() => {
    setNeedsProfileCompletion(false);
    refreshUser().catch((error) => {
      console.error('プロフィール再取得エラー:', error);
    });
  }, [refreshUser]);

  const value: AuthContextType = React.useMemo(
    () => ({
      user,
      student,
      role,
      isLoading,
      needsProfileCompletion,
      signIn,
      signUp,
      signOut,
      refreshUser,
      updateProfile,
      switchRole,
      completeProfile,
    }),
    [
      user,
      student,
      role,
      isLoading,
      needsProfileCompletion,
      signIn,
      signUp,
      signOut,
      refreshUser,
      updateProfile,
      switchRole,
      completeProfile,
    ],
  );

  return <AuthContext.Provider value={value}>{children}</AuthContext.Provider>;
}
