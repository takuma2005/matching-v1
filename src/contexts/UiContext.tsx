import React, { createContext, useContext, useState, ReactNode } from 'react';

interface UiState {
  homeScrollY: number;
  setHomeScrollY: (y: number) => void;
}

const UiContext = createContext<UiState | undefined>(undefined);

export function UiProvider({ children }: { children: ReactNode }) {
  const [homeScrollY, setHomeScrollY] = useState(0);

  const value = React.useMemo(
    () => ({ homeScrollY, setHomeScrollY }),
    [homeScrollY, setHomeScrollY],
  );

  return <UiContext.Provider value={value}>{children}</UiContext.Provider>;
}

export function useUi() {
  const ctx = useContext(UiContext);
  if (!ctx) throw new Error('useUi must be used within UiProvider');
  return ctx;
}
