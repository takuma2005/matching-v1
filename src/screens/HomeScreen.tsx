import { useFocusEffect, type NavigationProp, type ParamListBase } from '@react-navigation/native';
import { StackNavigationProp } from '@react-navigation/stack';
import React, { useCallback } from 'react';
import { View, Text, StyleSheet, ScrollView } from 'react-native';

import { HomeStackParamList } from '../navigation/HomeStackNavigator';
import { colors, spacing, typography, borderRadius } from '../styles/theme';

import BlurHeader from '@/components/home/BlurHeader';
import TutorsSection from '@/components/home/TutorsSection';
import UpcomingLessonCard from '@/components/home/UpcomingLessonCard';
import WelcomeCard from '@/components/home/WelcomeCard';
import { useHomeData } from '@/hooks/useHomeData';
type HomeScreenNavigationProp = StackNavigationProp<HomeStackParamList, 'HomeMain'>;

type Props = {
  navigation: HomeScreenNavigationProp;
};

export default function HomeScreen({ navigation }: Props) {
  const { recommendedTutors, newTutors, upcoming } = useHomeData();
  const [scrollY, setScrollY] = React.useState(0);


  const handleTutorPress = (tutorId: string) => {
    navigation.navigate('TutorDetail', { tutorId });
  };

  return (
    <View style={styles.root}>
      {/* Background ScrollView */}
      <ScrollView
        style={styles.scrollView}
        contentContainerStyle={styles.scrollContentContainer}
        showsVerticalScrollIndicator={false}
        onScroll={(e) => {
          const y = e.nativeEvent.contentOffset.y;
          setScrollY(y);
        }}
        scrollEventThrottle={16}
      >
        {/* Content with top padding for header */}
        <View style={styles.contentContainer}>
          <WelcomeCard name={user?.name || 'ゲスト'} />
          {/* 授業の予定（余白をやや詰める） */}
          <View style={[styles.section, styles.scheduleSection]}>
            <View style={[styles.sectionHeader, styles.sectionHeaderTight]}>
              <Text style={styles.sectionTitle}>授業の予定</Text>
            </View>
            <UpcomingLessonCard
              upcoming={upcoming}
              onPressDetail={() =>
                (navigation as unknown as NavigationProp<ParamListBase>).navigate('Lesson')
              }
            />
          </View>

          {/* Tutors Section: おすすめの先輩 */}
          <TutorsSection
            title="おすすめの先輩"
            tutors={recommendedTutors}
            onPressTutor={handleTutorPress}
            onToggleFavorite={(id) => {
            }}
          />

          {/* Tutors Section: 新着の先輩 */}
          <TutorsSection
            title="新着の先輩"
            tutors={newTutors}
            onPressTutor={handleTutorPress}
            onToggleFavorite={(id) => {
            }}
          />

          <View style={styles.bottomSpacing} />
        </View>
      </ScrollView>

      {/* Blur Header */}
      <BlurHeader
        coins={user?.coins ?? 0}
        onPressCoinManagement={() => navigation.navigate('CoinManagement')}
        onPressNotification={() => navigation.navigate('Notification')}
        scrollY={scrollY}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  root: {
    flex: 1,
    backgroundColor: '#F6FAFF',
  },
  fixedHeader: {
    paddingHorizontal: spacing.lg,
    paddingVertical: spacing.md,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    backgroundColor: colors.white,
    borderBottomWidth: 1,
    borderBottomColor: colors.gray200,
    zIndex: 1000,
  },
  scrollView: {
    flex: 1,
    backgroundColor: '#F6FAFF',
  },
  scrollContentContainer: {
    paddingTop: 140, // SafeArea + ヘッダー分のスペース（より大きく）
    paddingBottom: 100, // タブバー分のスペース
  },
  headerGradient: {
    paddingTop: spacing.md,
    paddingBottom: spacing.xl,
  },
  appName: {
    fontSize: typography.sizes?.h3 || 20,
    fontWeight: '700',
    color: colors.primary,
  },
  headerRight: {
    flexDirection: 'row',
    alignItems: 'center',
    gap: spacing.sm,
  },
  headerCoinButton: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: colors.gray100,
    paddingHorizontal: spacing.md,
    height: 32,
    borderRadius: borderRadius.full,
    marginRight: spacing.sm,
  },
  headerCoinText: {
    marginLeft: spacing.xs / 2,
    color: colors.gray800,
    fontWeight: '700',
    fontSize: typography.sizes?.caption || 12,
  },
  notificationButton: {
    padding: spacing.xs,
    borderRadius: borderRadius.full,
    backgroundColor: colors.gray100,
    width: 32,
    height: 32,
    alignItems: 'center',
    justifyContent: 'center',
  },
  welcomeSection: {
    paddingHorizontal: spacing.lg,
    paddingVertical: spacing.lg,
  },
  welcomeText: {
    fontSize: typography.sizes?.h3 || 20,
    fontWeight: '700',
    color: colors.white,
    marginBottom: spacing.xs,
  },
  subtitle: {
    fontSize: typography.sizes?.body || 16,
    color: 'rgba(255,255,255,0.9)',
  },
  contentContainer: {
    backgroundColor: '#F6FAFF',
    flex: 1,
  },
  overlaidCardSection: {
    paddingHorizontal: spacing.lg,
    marginBottom: spacing.lg,
  },
  coinCard: {
    backgroundColor: colors.white,
    borderRadius: borderRadius.xl,
    padding: spacing.lg,
    flexDirection: 'row',
    alignItems: 'center',
    shadowColor: colors.black,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 8,
    elevation: 4,
  },
  coinIconContainer: {
    width: 40,
    height: 40,
    backgroundColor: colors.warning + '20',
    borderRadius: 20,
    alignItems: 'center',
    justifyContent: 'center',
    marginRight: spacing.md,
  },
  coinInfo: {
    flex: 1,
  },
  coinLabel: {
    fontSize: typography.sizes?.caption || 12,
    color: colors.gray600,
    marginBottom: 2,
  },
  coinAmount: {
    fontSize: typography.sizes?.h3 || 20,
    fontWeight: '700',
    color: colors.gray900,
  },
  addCoinButton: {
    backgroundColor: colors.primary,
    paddingHorizontal: spacing.lg,
    paddingVertical: spacing.sm,
    borderRadius: borderRadius.full || 999,
  },
  addCoinText: {
    fontSize: typography.sizes?.caption || 12,
    color: colors.white,
    fontWeight: '600',
  },
  lessonCard: {
    backgroundColor: colors.white,
    marginHorizontal: spacing.md,
    borderRadius: borderRadius.lg,
    padding: spacing.sm,
    marginBottom: spacing.sm,
    shadowColor: colors.black,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.08,
    shadowRadius: 6,
    elevation: 3,
  },
  lessonHeader: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: spacing.xs / 2,
    gap: spacing.xs,
  },
  lessonTitle: {
    fontSize: typography.sizes?.caption || 12,
    fontWeight: '700',
    color: colors.gray900,
    marginLeft: spacing.xs / 2,
  },
  lessonMeta: {
    fontSize: typography.sizes?.caption || 12,
    color: colors.gray600,
    marginBottom: spacing.xs / 4,
  },
  lessonTutor: {
    fontSize: typography.sizes?.caption || 12,
    color: colors.gray700,
    marginBottom: spacing.xs / 2,
  },
  lessonRow: {
    flexDirection: 'row',
    alignItems: 'center',
    gap: spacing.sm,
  },
  lessonDetailButton: {
    backgroundColor: colors.primary,
    paddingHorizontal: spacing.md,
    height: 28,
    minWidth: 96,
    borderRadius: borderRadius.md,
    alignItems: 'center',
    justifyContent: 'center',
  },
  lessonDetailButtonText: {
    color: colors.white,
    fontWeight: '700',
    fontSize: typography.sizes?.caption || 12,
  },
  emptyLessonText: {
    marginHorizontal: spacing.lg,
    color: colors.gray600,
    fontSize: typography.sizes?.caption || 12,
  },
  sectionGap: {
    marginBottom: spacing.md,
  },
  scheduleSection: {
    paddingVertical: spacing.sm,
  },
  sectionHeaderTight: {
    marginBottom: spacing.sm + spacing.xs,
  },

  quickActionCard: {
    backgroundColor: colors.white,
    marginHorizontal: spacing.md,
    borderRadius: borderRadius.xl,
    padding: spacing.lg,
    marginBottom: spacing.xl,
    shadowColor: colors.black,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 8,
    elevation: 4,
  },
  quickActions: {
    flexDirection: 'row',
    justifyContent: 'space-around',
  },
  quickActionItem: {
    alignItems: 'center',
    flex: 1,
  },
  quickActionIcon: {
    width: 48,
    height: 48,
    backgroundColor: colors.primary + '15',
    borderRadius: 12,
    alignItems: 'center',
    justifyContent: 'center',
    marginBottom: spacing.sm,
  },
  quickActionText: {
    fontSize: typography.sizes?.caption || 12,
    color: colors.gray700,
    textAlign: 'center',
    fontWeight: '500',
  },
  section: {
    paddingVertical: spacing.sm,
    backgroundColor: 'transparent',
  },
  sectionHeader: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: spacing.md,
    marginBottom: spacing.sm + spacing.xs,
  },
  sectionTitle: {
    fontSize: typography.sizes?.h4 || 18,
    fontWeight: '600',
    color: colors.gray900,
  },
  seeAllButton: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  seeAllText: {
    fontSize: typography.sizes?.caption || 12,
    color: colors.primary,
    marginRight: spacing.xs / 2,
  },
  tutorCardWrapper: {
    position: 'relative',
    marginHorizontal: 0,
    marginBottom: 0,
  },
  tutorCard: {
    backgroundColor: colors.white,
    borderRadius: borderRadius.lg,
    borderWidth: 1,
    borderColor: colors.gray200,
    shadowColor: colors.black,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.08,
    shadowRadius: 4,
    elevation: 2,
  },
  tutorCardContent: {
    flexDirection: 'row',
    padding: spacing.lg,
  },
  tutorAvatar: {
    width: 50,
    height: 50,
    borderRadius: 25,
    alignItems: 'center',
    justifyContent: 'center',
    marginRight: spacing.md,
    position: 'relative',
  },
  tutorAvatarText: {
    fontSize: typography.sizes?.h4 || 18,
    fontWeight: '700',
    color: colors.white,
  },
  onlineDot: {
    position: 'absolute',
    bottom: 2,
    right: 2,
    width: 12,
    height: 12,
    borderRadius: 6,
    backgroundColor: colors.success,
    borderWidth: 2,
    borderColor: colors.white,
  },
  tutorInfo: {
    flex: 1,
  },
  tutorHeader: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: spacing.xs,
  },
  tutorName: {
    fontSize: typography.sizes?.body || 16,
    fontWeight: '600',
    color: colors.gray900,
  },
  ratingContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  ratingText: {
    fontSize: typography.sizes?.caption || 12,
    color: colors.gray600,
    marginLeft: 2,
  },
  tutorSchool: {
    fontSize: typography.sizes?.caption || 12,
    color: colors.gray600,
    marginBottom: spacing.sm,
  },
  tutorSubjects: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    marginBottom: spacing.sm,
    gap: spacing.xs,
  },
  subjectChip: {
    backgroundColor: colors.primary + '15',
    paddingHorizontal: spacing.sm,
    paddingVertical: 4,
    borderRadius: borderRadius.full || 999,
  },
  subjectChipText: {
    fontSize: 10,
    color: colors.primary,
    fontWeight: '500',
  },
  tutorFooter: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  tutorRate: {
    fontSize: typography.sizes?.body || 16,
    fontWeight: '600',
    color: colors.primary,
  },
  favoriteButtonHome: {
    position: 'absolute',
    bottom: spacing.sm,
    right: spacing.sm,
    backgroundColor: colors.white,
    borderRadius: borderRadius.full || 999,
    width: 32,
    height: 32,
    alignItems: 'center',
    justifyContent: 'center',
    shadowColor: colors.black,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.15,
    shadowRadius: 4,
    elevation: 3,
    borderWidth: 1,
    borderColor: colors.gray100,
  },
  bottomSpacing: {
    height: spacing.xl,
    backgroundColor: '#F6FAFF',
  },
});
