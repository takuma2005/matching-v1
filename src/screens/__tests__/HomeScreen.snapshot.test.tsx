import { render } from '@testing-library/react-native';
import React from 'react';
/* eslint-disable @typescript-eslint/no-explicit-any */
import { SafeAreaProvider } from 'react-native-safe-area-context';

import HomeScreen from '../HomeScreen';

import { AuthProvider } from '@/contexts/AuthContext';
import { FavoritesProvider } from '@/contexts/FavoritesContext';
import { UserProvider } from '@/contexts/UserContext';

const mockNavigation = { navigate: jest.fn() };

describe('HomeScreen snapshot', () => {
  it('matches snapshot', () => {
    const { toJSON } = render(
      <SafeAreaProvider>
        <AuthProvider>
          <UserProvider>
            <FavoritesProvider>
              <HomeScreen navigation={mockNavigation as any} />
            </FavoritesProvider>
          </UserProvider>
        </AuthProvider>
      </SafeAreaProvider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
