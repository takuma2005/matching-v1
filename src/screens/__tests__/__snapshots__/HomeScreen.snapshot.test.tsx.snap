// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen snapshot matches snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#F6FAFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "paddingBottom": 100,
        "paddingTop": 140,
      }
    }
    onScroll={[Function]}
    scrollEventThrottle={16}
    showsVerticalScrollIndicator={false}
    style={
      {
        "backgroundColor": "#F6FAFF",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          {
            "backgroundColor": "#F6FAFF",
            "flex": 1,
          }
        }
      >
        <View
          style={
            {
              "borderRadius": 16,
              "elevation": 8,
              "marginBottom": 16,
              "marginHorizontal": 16,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 8,
            }
          }
          testID="welcome-card"
        >
          <LinearGradient
            colors={
              [
                "#8EC5FF",
                "#60A5FA",
                "#38BDF8",
              ]
            }
            end={
              {
                "x": 1,
                "y": 1,
              }
            }
            start={
              {
                "x": 0,
                "y": 0,
              }
            }
            style={
              {
                "borderRadius": 16,
                "paddingHorizontal": 24,
                "paddingVertical": 24,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#ffffff",
                  "fontSize": 20,
                  "fontWeight": "700",
                  "marginBottom": 4,
                }
              }
            >
              おはよう、
              ゲスト
              さん！
            </Text>
            <Text
              style={
                {
                  "color": "rgba(255,255,255,0.9)",
                  "fontSize": 16,
                }
              }
            >
              👋 今日も新しい学びの出会いを見つけよう
            </Text>
          </LinearGradient>
        </View>
        <View
          style={
            [
              {
                "backgroundColor": "transparent",
                "paddingVertical": 8,
              },
              {
                "paddingVertical": 8,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 12,
                  "paddingHorizontal": 16,
                },
                {
                  "marginBottom": 12,
                },
              ]
            }
          >
            <Text
              style={
                {
                  "color": "#111827",
                  "fontSize": 18,
                  "fontWeight": "600",
                }
              }
            >
              授業の予定
            </Text>
          </View>
          <Text
            style={
              {
                "color": "#4b5563",
                "fontSize": 12,
                "marginHorizontal": 24,
              }
            }
          >
            次の授業はまだありません
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "transparent",
              "paddingVertical": 8,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 12,
                "paddingHorizontal": 16,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#111827",
                  "fontSize": 18,
                  "fontWeight": "600",
                }
              }
            >
              おすすめの先輩
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "backgroundColor": "transparent",
              "paddingVertical": 8,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 12,
                "paddingHorizontal": 16,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#111827",
                  "fontSize": 18,
                  "fontWeight": "600",
                }
              }
            >
              新着の先輩
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "backgroundColor": "#F6FAFF",
              "height": 32,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      [
        {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 1000,
        },
        {
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "backgroundColor": "#ffffff",
          },
        ]
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "#3b82f6",
            "fontSize": 20,
            "fontWeight": "700",
          }
        }
      >
        センパイ
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#f59e0b20",
              "borderRadius": 9999,
              "flexDirection": "row",
              "height": 28,
              "marginRight": 8,
              "opacity": 1,
              "paddingHorizontal": 12,
            }
          }
          testID="header-coin-button"
        >
          <MaterialIcons
            color="#f59e0b"
            name="paid"
            size={16}
          />
          <Text
            style={
              {
                "color": "#f59e0b",
                "fontSize": 12,
                "fontWeight": "700",
                "marginLeft": 4,
              }
            }
          >
            0
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#f3f4f6",
              "borderRadius": 9999,
              "height": 32,
              "justifyContent": "center",
              "opacity": 1,
              "padding": 4,
              "width": 32,
            }
          }
          testID="notification-icon"
        >
          <MaterialIcons
            color="#374151"
            name="notifications"
            size={20}
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#e5e7eb",
          "height": 0.5,
        }
      }
    />
  </View>
</View>
`;
